src/battle_util.c

```diff
@@ -4865,7 +4865,7 @@ u8 ItemBattleEffects(u8 caseID, u8 battlerId, bool8 moveTurn)
         {
         case HOLD_EFFECT_TOXIC_ORB:
             if (!gBattleMons[battlerId].status1
-                && !IS_BATTLER_OF_TYPE(battlerId, TYPE_POISON) && !IS_BATTLER_OF_TYPE(battlerId, TYPE_STEEL)
+                && CanPoisonType(battlerId, gEffectBattler)
```

Comment:
I don't actually think Corrosion applies to Toxic Orb, so I would just leave the type checks theere

Reply:
It does according to Bulbapedia but haven't tested it in-game.

Reply:
Found this video of somebody testing this in SM and indeed Toxic Orb can poison if the holder has Corrosion.
https://www.youtube.com/watch?v=IjFHFBXUpV8
Interestingly, they also tested Magic Bounce's effect and the bounced-back attack can't poison even if it was originally by a Corrosion owner

Reply:
Alright then, shouldn't it be CanPoisonType(battlerId, battlerId) then?

Reply:
Tested and worked both ways. Do I just leave it as (battlerId, battlerId)?